File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Somalia.m
//
//  UIImage+Somalia.m
//  CountriesFlags
//

#import "Countries/UIImage+Somalia.h"

#import "Extensions/UIBezierPath+Star.h"
#import "Extensions/CGSize+Extension.h"

@implementation UIImage (Somalia)

+ (UIImage * _Nullable)flagSomaliaWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:65.0f/255.0f
                                         green:143.0f/255.0f
                                          blue:222.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    [[UIBezierPath starAtPoint:CGSizeMidPoint(size)
                        radius:size.height / 4.0f
                          type:FlagStarTypeFivePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Somalia.swift
//
//  UIImage+Somalia.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Somalia "wikipedia"
     
     Generates the flag of Somalia.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Somalia.
     */
    internal static func flagSomalia(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 65.0 / 255.0,
                                green: 143.0 / 255.0,
                                blue: 222.0 / 255.0,
                                alpha: 1.0)
        let colorWhite = UIColor.white

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBlue.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIBezierPath.star(at: size.midPoint, radius: size.height / 4.0).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
